# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpChannelChild.cpp
# Commit: effee4ab270c
# Full Hash: effee4ab270cba85ad53b7be844212bc458ec1a6
# Author: Honza Bambas <honzab.moz@firemni.cz>
# Date: 2019-10-03 04:38:57
# Regressor Bug: 1584207
# File Overlap Count: 1
# Description:
#   Bug 1584207 - Forward nsILoadInfo.requestBlockingReason to the parent process for both old and new redirect channels, r=dragana,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D47264
# ==============================================================================

diff -r a49132352f5b -r effee4ab270c netwerk/protocol/http/HttpChannelChild.cpp
--- a/netwerk/protocol/http/HttpChannelChild.cpp	Wed Oct 02 20:53:50 2019 +0200
+++ b/netwerk/protocol/http/HttpChannelChild.cpp	Tue Oct 01 16:50:23 2019 +0000
@@ -2384,19 +2384,25 @@
     appCacheChannel->GetChooseApplicationCache(&chooseAppcache);
   }
 
+  uint32_t sourceRequestBlockingReason = 0;
+  if (mLoadInfo) {
+    mLoadInfo->GetRequestBlockingReason(&sourceRequestBlockingReason);
+  }
+
   nsCOMPtr<nsILoadInfo> newChannelLoadInfo = nullptr;
   nsCOMPtr<nsIChannel> newChannel = do_QueryInterface(mRedirectChannelChild);
   if (newChannel) {
     newChannelLoadInfo = newChannel->LoadInfo();
   }
 
-  ChildLoadInfoForwarderArgs loadInfoForwarder;
-  LoadInfoToChildLoadInfoForwarder(newChannelLoadInfo, &loadInfoForwarder);
+  ChildLoadInfoForwarderArgs targetLoadInfoForwarder;
+  LoadInfoToChildLoadInfoForwarder(newChannelLoadInfo,
+                                   &targetLoadInfoForwarder);
 
   if (CanSend())
-    SendRedirect2Verify(result, *headerTuples, loadInfoForwarder, loadFlags,
-                        referrerInfo, redirectURI, corsPreflightArgs,
-                        chooseAppcache);
+    SendRedirect2Verify(result, *headerTuples, sourceRequestBlockingReason,
+                        targetLoadInfoForwarder, loadFlags, referrerInfo,
+                        redirectURI, corsPreflightArgs, chooseAppcache);
 
   return NS_OK;
 }